src/librustdoc/clean/auto_trait.rs

```diff
@@ -315,12 +315,12 @@ impl<'a, 'tcx> AutoTraitFinder<'a, 'tcx> {
         tcx: TyCtxt<'tcx>,
         pred: ty::Predicate<'tcx>,
     ) -> FxHashSet<GenericParamDef> {
-        let regions = match pred.kind() {
-            ty::PredicateKind::Trait(poly_trait_pred, _) => {
-                tcx.collect_referenced_late_bound_regions(&poly_trait_pred)
+        let regions = match pred.skip_binders() {
```

Comment:
what about writing this as

```rust
let regions = match pred.skip_binders() {
    ty::PredicateAtom::Trait(..) | ty::PredicateAtom::Projection(..) => tcx.collect_referenced_late_bound_regions(&pred),
    _ => return FxHashSet::default(),
};
```

isn't that equivalent?

Reply:
it is :thinking: but is it cleaner? I had to think for a few seconds on why this is correct :sweat_smile:

Reply:
To me it is cleaner, but YMMV. To be honest, I'm not even sure why the match exists -- but the idea is that we want to "Collect the referenced late-bound regions from the predicate, at least for some predicates"